File: app/src/main/java/com/mikhailovskii/trakttv/ui/movie_detail/MovieDetailContract.java
package com.mikhailovskii.trakttv.ui.movie_detail;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.mikhailovskii.trakttv.data.entity.Movie;
import com.mikhailovskii.trakttv.ui.base.MvpPresenter;
import com.mikhailovskii.trakttv.ui.base.MvpView;

public interface MovieDetailContract {

    interface MovieDetailView extends MvpView {

        void onMovieDetailsLoaded(@NonNull Movie movie);

        void onMovieDetailsFailed();

        void onMoviesAdded();

        void onMoviesAddingFailed();

    }

    interface MovieDetailPresenter extends MvpPresenter<MovieDetailView> {

        void loadMovieDetails(@Nullable String slugId);

        void addMovieToFavorites(@NonNull Bundle bundle);

    }

}

File: app/src/main/java/com/mikhailovskii/trakttv/ui/movie_detail/MovieDetailContract.kt
package com.mikhailovskii.trakttv.ui.movie_detail

import com.mikhailovskii.trakttv.data.entity.Movie
import com.mikhailovskii.trakttv.ui.base.MvpPresenter
import com.mikhailovskii.trakttv.ui.base.MvpView

interface MovieDetailContract {

    interface MovieDetailView : MvpView {

        fun onMovieDetailsLoaded(movie: Movie)

        fun onMovieDetailsFailed()

        fun onMoviesAdded()

        fun onMoviesAddingFailed()

        fun onMovieRemoved()

        fun onMovieRemoveFailed()

    }

    interface MovieDetailPresenter : MvpPresenter<MovieDetailView> {

        fun loadMovieDetails(slugId: String?)

        fun addMovieToFavorites(movie: Movie)

        fun removeMovieFromFavorites(name:String)

    }

}
